### PR TITLE
add roles and other collection things

### PR DESCRIPTION
**Pull request by thedoubl3j**
first pass adding roles and basic collection things.

### CODE REVIEW

File: roles/applications/meta/argument_specs.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+argument_specs:
+  main:
+    short_description: An Ansible Role to create Applications on Ansible gateway.
+    options:
+      gateway_applications:
+        description: Data structure describing your applications
+        type: list
+        required: True
```
Comment (djdanielsson) [inline roles/applications/meta/argument_specs.yml:9]: Let's be consistent either go with True or true but don't randomly use both

File: galaxy.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# A short summary description of the collection
+description: A collection of roles to manage Ansible Automation Platform 
+
+# Either a single license or a list of licenses for content inside of a collection. Ansible Galaxy currently only
+# accepts L(SPDX,https://spdx.org/licenses/) licenses. This key is mutually exclusive with 'license_file'
+license:
+
+# The path to the license file for the collection. This path is relative to the root of the collection. This key is
+# mutually exclusive with 'license'
+license_file: 'COPYING'
```
Comment (djdanielsson) [inline galaxy.yml:33]: Is this correct?

File: roles/authenticator_maps/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  author: "Martin Slemr"
+  description: "An Ansible Role to create authenticator maps in automation platform gateway."
+  company: "Red Hat"
+
+  # If the issue tracker for your role is not on github, uncomment the
+  # next line and provide a value
+  # issue_tracker_url: http://example.com/issue/tracker
+  license: "GPLv3"
+
+  min_ansible_version: "2.14"
```
Comment (djdanielsson) [inline roles/authenticator_maps/meta/main.yml:13]: Need to be consistent with min Ansible version, also I think the technically correct way to pick a version is without quotes and be 3 levels like 2.16.0

**Review by sean-m-sullivan — APPROVED** [review]
As david pointed out we have more to review

File: roles/authenticators/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+## Data Structure
+
+### Authenticator Arguments
+
+Options for the `gateway_authenticators` variable:
+
+| Variable Name    |    Default Value    | Required | Type | Description                                                                                                                  |
+|:-----------------|:-------------------:|:--------:|:----:|:-----------------------------------------------------------------------------------------------------------------------------|
+| `name`           |         N/A         |   yes    | str  | The name of the resource                                                                                                     |
```
Comment (djdanielsson) [inline roles/authenticators/README.md:29]: Do not use yes but true or false

File: roles/routes/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+- name: "Routes | Configuration"  # noqa fqcn[action-core]
+  ansible.gateway_configuration.route:
```
Comment (djdanielsson) [inline roles/routes/tasks/main.yml:3]: Just verifying this is correct FQCN

**Review by djdanielsson — CHANGES_REQUESTED** [review]
I'm on mobile but skimming through I saw a few things that needed changing

**Review by djdanielsson — APPROVED** [review]
Approving and I'll fix the stuff when I am back online

**Pull request merged**